src/scheduler/scheduler.rs

```diff
@@ -553,22 +553,16 @@ impl<VM: VMBinding> GCWorkScheduler<VM> {
         // USDT tracepoint for the end of GC.
         probe!(mmtk, gc_end);
 
-        #[cfg(feature = "count_live_bytes_in_gc")]
-        {
-            let live_bytes = mmtk.state.get_live_bytes_in_last_gc();
-            let used_bytes =
-                mmtk.get_plan().get_used_pages() << crate::util::constants::LOG_BYTES_IN_PAGE;
-            debug_assert!(
-                live_bytes <= used_bytes,
-                "Live bytes of all live objects ({} bytes) is larger than used pages ({} bytes), something is wrong.",
-                live_bytes, used_bytes
-            );
-            info!(
-                "Live objects = {} bytes ({:04.1}% of {} used pages)",
-                live_bytes,
-                live_bytes as f64 * 100.0 / used_bytes as f64,
-                mmtk.get_plan().get_used_pages()
-            );
+        if *mmtk.get_options().count_live_bytes_in_gc {
+            for (space_name, &stats) in mmtk.state.live_bytes_in_last_gc.borrow().iter() {
+                info!(
+                    "{} = {} bytes ({:.1}% of {} used pages)",
```

Comment:
In the previous `info!` log, we print "End of GC" and the number of pages.  But here we print the number of bytes.  To be consistent, and for the convenience of the users, I suggest we print both number of pages and the number of bytes in both log statements.  The result will look like the following:

```
[INFO] End of GC (1000/1024 pages, 4000/4096 KiB, took 100 ms)
[INFO]   nursery = 0/400 pages, 0/1600 KiB (0% of 1000 used pages)
[INFO]   immixspace = 500/600 pages, 2000/2400 KiB (50% of 1000 used pages)
[INFO]   ...
```

And it should be OK to use KiB (1 KiB = 1024 B) instead of bytes, or KB for short, although in SI units 1 KB = 1000 B.  Since we count at page granularity, the result will always be a integer multiple of 4096 bytes.

Reply:
I changed the log a bit, but I did not log bytes for everything -- I think it would be too verbose, and it is redundant info (pages and bytes can be computed from each other anyway).

Reply:
> I changed the log a bit, but I did not log bytes for everything -- I think it would be too verbose, and it is redundant info (pages and bytes can be computed from each other anyway).

I think you missed the point.  The log information is for *humans* to read when the program is executing.  Although pages and bytes can be converted to each other, humans are not that good at multiplying a number by 4 (to get KiB) or by 4096 (to get bytes).  The log should let the user know the amount of used/total bytes and used/total pages in a glance.  That's what developers usually do when debugging.  They look at the program output and want to know the number immediately without having to open a Python shell and type something like (`500 * 4096`) to compute the number of bytes, or `1050456 / 4096` to get the number of pages.

Reply:
The percentage is what matters. Human only need to read the percentage. Other info just shows how the percentage is calculated. Presenting live bytes as pages, or presenting used pages as bytes can be misleading.

We can do this:
```
[INFO] End of GC (1000/1024 pages, took 100 ms)
[INFO]   nursery = 400 pages (0% live)
[INFO]   immixspace = 600 pages (50% live)
```